File: python/chatGPT/Builder/round1/Appointment.py

class Appointment:
    def __init__(self, doctor_name, patient_name, appointment_type, date_time, note, creator_name, clinic_contact):
        self.doctor_name = doctor_name
        self.patient_name = patient_name
        self.appointment_type = appointment_type
        self.date_time = date_time
        self.note = note
        self.creator_name = creator_name
        self.clinic_contact = clinic_contact

    def __str__(self):
        return (f"Appointment with Dr. {self.doctor_name} for {self.patient_name} "
                f"on {self.date_time} for {self.appointment_type}. "
                f"Note: {self.note}. "
                f"Appointment created by {self.creator_name}. "
                f"Contact: {self.clinic_contact}")

File: java/BuilderPattern/src/main/java/org/chatgpt/round3/Appointment.java
package org.chatgpt.round3;
import java.time.LocalDateTime;


public class Appointment {

    private final String patientName;
    private final String doctorName;
    private final AppointmentType appointmentType;
    private final String bookedBy;
    private final String clinicContact;
    private final LocalDateTime appointmentDateTime;
    private final String notes;

    private Appointment(Builder builder) {
        this.patientName = builder.patientName;
        this.doctorName = builder.doctorName;
        this.appointmentType = builder.appointmentType;
        this.bookedBy = builder.bookedBy;
        this.clinicContact = builder.clinicContact;
        this.appointmentDateTime = builder.appointmentDateTime;
        this.notes = builder.notes;
    }

    public static class Builder {
        private final String patientName;
        private final String doctorName;
        private final AppointmentType appointmentType;
        private final String bookedBy;
        private final String clinicContact;
        private final LocalDateTime appointmentDateTime;
        private String notes;

        public Builder(String patientName, String doctorName, AppointmentType appointmentType,
                       String bookedBy, String clinicContact, LocalDateTime appointmentDateTime) {
            this.patientName = patientName;
            this.doctorName = doctorName;
            this.appointmentType = appointmentType;
            this.bookedBy = bookedBy;
            this.clinicContact = clinicContact;
            this.appointmentDateTime = appointmentDateTime;
        }

        public Builder notes(String notes) {
            this.notes = notes;
            return this;
        }

        public Appointment build() {
            return new Appointment(this);
        }
    }

    // Getters for all fields

    public String getPatientName() {
        return patientName;
    }

    public String getDoctorName() {
        return doctorName;
    }

    public AppointmentType getAppointmentType() {
        return appointmentType;
    }

    public String getBookedBy() {
        return bookedBy;
    }

    public String getClinicContact() {
        return clinicContact;
    }

    public LocalDateTime getAppointmentDateTime() {
        return appointmentDateTime;
    }

    public String getNotes() {
        return notes;
    }
}

// Enum for Appointment Type
enum AppointmentType {
    TOOTH_EXTRACTION,
    DEVICE_INSTALLATION,
    MONTHLY_CHECKUP,
    DEVICE_REMOVAL
}